python/ray/external_storage.py

```diff
@@ -60,6 +60,8 @@ def parse_url_with_offset(url_with_offset: str) -> Tuple[str, int, int]:
     query_dict = urllib.parse.parse_qs(parsed_result.query)
     # Split by ? to remove the query from the url.
     base_url = parsed_result.geturl().split("?")[0]
+    if "offset" not in query_dict or "size" not in query_dict:
```

Comment:
Isn't this supposed to not happen ever? Why don't we use assert here?

Reply:
This gives a better error in case of mis-formatted URL (including the origin URL)